client/my-sites/themes/helpers.js

```diff
@@ -148,20 +149,45 @@ export function interlaceThemes( wpComThemes, wpOrgThemes, searchTerm, isLastPag
 		  )
 		: [];
 
-	const matchingTheme = wpComThemes.find( isMatchingTheme );
+	const interlacedThemes = [];
+
+	// 1. Exact match.
+	const matchingTheme =
+		wpComThemes.find( isMatchingTheme ) || validWpOrgThemes.find( isMatchingTheme );
+	if ( matchingTheme ) {
+		interlacedThemes.push( matchingTheme );
+	}
+
+	// 2. WP.com themes.
 	const restWpComThemes = matchingTheme
 		? wpComThemes.filter( ( theme ) => theme.id !== matchingTheme.id )
 		: wpComThemes;
-	const matchingWpOrgTheme = validWpOrgThemes.find( isMatchingTheme );
-	const restWpOrgThemes = matchingWpOrgTheme
-		? validWpOrgThemes.filter( ( theme ) => theme.id !== matchingWpOrgTheme.id )
-		: validWpOrgThemes;
-
-	return [
-		...( matchingTheme ? [ matchingTheme ] : [] ),
-		...( matchingWpOrgTheme ? [ matchingWpOrgTheme ] : [] ),
-		...restWpComThemes,
-		// Include WP.org themes after the last page of WP.com themes.
-		...( isEnabled( 'themes/interlaced-dotorg-themes' ) && isLastPage ? restWpOrgThemes : [] ),
-	];
+	interlacedThemes.push( ...restWpComThemes );
+
+	// 3. WP.org themes (only if the list of WP.com themes has reached the last page).
+	if ( isEnabled( 'themes/interlaced-dotorg-themes' ) && isLastPage ) {
+		// 3.1. WP.org block themes.
+		const restWpOrgBlockThemes = validWpOrgThemes.filter(
```

Comment:
What about extracting the "rest of Dotorg themes" once before checking if they are block or classic, as we do for Dotcom ones?

E.g.

```js
const restWpOrgThemes = matchingTheme
	? wpOrgThemes.filter( ( theme ) => theme.id !== matchingTheme.id )
	: wpOrgThemes;

const restWpOrgBlockThemes = restWpOrgThemes.filter( ... );
const restWpOrgClassicThemes = restWpOrgThemes.filter( ... );
```

Reply:
That's how I had it at some point and ultimately decided to remove the `restWpOrgThemes` var, because that way we only need to iterate `wpOrgThemes` twice instead of three times. 

It's probably a micro-optimization that doesn't have too much impact, so happy to bring it back.

Reply:
Not needed anymore after using the single loop you suggested.